[RFC] Repeatable directives (#472)
graphql/graphql-spec: spec/Section 3 -- Type System.md

}
```

Directive may be defined as repeatable per location with the `repeatable` keyword:

```graphql example
directive @example repeatable on OBJECT | INTERFACE
```

> 

Reviewer: Would be great to include a more complex example that illustrates how `repeatable` is useful or at least add a note that directive order is significant and you can use it to create chains of directives.

Author: I extended the example https://github.com/facebook/graphql/commit/218e0da6d614bc5be33bc4a6378618ad4a59ea51 and the description https://github.com/facebook/graphql/commit/477919e3251c3b7852a2ec21990bbdf881ff1553.

I struggled with it a bit since I find it hard to strike the right balance between completeness of the example including its description and succinct focused content required from a perspective of the specification. For this reason I also hesitated mentioning the order significance since it has its own dedication section with a separate example.

WDYT?

}
```

A directive may be defined as repeatable at any permitted location with the `repeatable` 

> keyword. Repeatable directives often useful when the same directive should be used with 

Reviewer 2: ```suggestion
keyword. Repeatable directives are often useful when the same directive should be used with 
```

> different arguments at the one location, especially in cases where additional information 

Reviewer 2: ```suggestion
different arguments at a single location, especially in cases where additional information 
```

> need to be provided in a form of directive via a type or a schema extension:

Reviewer 2: ```suggestion
needs to be provided to a type or schema extension through a directive:
```
